### PR TITLE
revise the litecoin docs

### PR DESCRIPTION
**Pull request by contrun**

### CODE REVIEW

File: docs/litecoin.md [hunk 1]
```diff
@@ -1,43 +1,99 @@
...
+wget -O "$tarball" https://download.litecoin.org/litecoin-0.21.2.2/linux/litecoin-0.21.2.2-x86_64-linux-gnu.tar.gz
+tar xvzf "$tarball"
+sudo cp -r litecoin-*/* /usr/local
 ```
 
-### Get the transaction hash
-You need to construct a skeleton transaction and obtain the transaction hash. See [Tutorial | Lumos](https://lumos-website.vercel.app/#5-create-a-transfer-transaction) for a example to do such thing with Lumos.
+## Starting the litecoin daemon
+Litecoin daemon is required in order to for litecoin-cli to work. For example,
+we can run `litecoind -testnet` to start a litecoin testnet node.
```
Comment (gpBlockchain) [inline docs/litecoin.md:26]:
```suggestion
To work with Litecoin in a test environment, you need to start a Litecoin testnet node. You can do this using the `litecoind` command with the `-testnet` option. Here's how:

```bash
litecoind -testnet -daemon
```
```

File: docs/litecoin.md [hunk 2]
```diff
@@ -1,43 +1,99 @@
...
+litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet importprivkey cQoJiU5ECnVpRqfV5dWKDE2sLQq6516Tja1Hb1GABUV24n7WkqV4 ckb-auth-test-privkey false
 ```
-cQoJiU5ECnVpRqfV5dWKDE2sLQq6516Tja1Hb1GABUV24n7WkqV4 1970-01-01T00:00:01Z label=ckb-auth-test-privkey # addr=mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2,QjpdvL4h5jnfaj1uV5ifJNUAYZTTbjgFH5,tltc1qrz8z67vtu38pq2yzqtq7unftmsaueq6a8da5n2,tmweb1qqfgxqndkjczrcv2k6gc0f8jmzsug358j2dpf8t0dxe4jaheu3swp5qemjfm7c2v7gt5sl5snf9kr6tygl3t773l6spt4cmuel4d92m038gkysq8d
+
+## Obtaining the litecoin address to sign data
+Due to the deprecation of [old address format](https://en.bitcoin.it/wiki/Technical_background_of_version_1_Bitcoin_addresses) ([upstream bitcoin did this](https://river.com/learn/terms/h/hd-wallet)),
+we need to obtain the legacy address of the litecoin account to sign messages.
+
+The easiest way to do such thing is to label the account (say with `ckb-auth-test-privkey`) 
+and dump all its address with `litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet getaddressesbylabel ckb-auth-test-privkey`
```
Comment (gpBlockchain) [inline docs/litecoin.md:50]:
```suggestion
and dump all its address,we can run:
 ```
 litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet getaddressesbylabel ckb-auth-test-privkey
```
And here's the output when I run that script:

```

File: docs/litecoin.md [hunk 3]
```diff
@@ -1,43 +1,99 @@
...
+([keis/base58](https://github.com/keis/base58) and [xxd(1)](https://linux.die.net/man/1/xxd)required).
 
-### Signing the message
-Signing is easier after obtained the wallet address and imported the private key. Running
-`SIGNATURE="$(litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet signmessage "$ADDRESS" "$MESSAGE")"` is enough.
-You can also verify the result with `litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet verifymessage "$ADDRESS" "$SIGNATURE" "$MESSAGE"`.
+## Signing a message with litecoin-cli
+To sign the message `29553f9e37fa16e45f1d3e616ac5366f6afd9936477f2d6fc870f49bdf540157`, we can run
+`litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet signmessage mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2 29553f9e37fa16e45f1d3e616ac5366f6afd9936477f2d6fc870f49bdf540157`,
+whose output `H2gWHSzffPkvD2QYD9jdLTuM5x7cax4E7Ax0Nm79T90wLU0LcK/DEYENggsC5G4FK/69qJP9eXp9zHkZ40PH6ME=` is the base64-encoded form of the signature. The binary form can be fed to ckb-auth for authentication.
```
Comment (gpBlockchain) [inline docs/litecoin.md:99]: Should we add how to use `ckb-auth` for verification in the contract into the document?

File: docs/litecoin.md [hunk 4]
```diff
@@ -1,43 +1,99 @@
...
+the command `base58 -d <<< mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2 | xxd -s 1 -l 20 -p` 
+([keis/base58](https://github.com/keis/base58) and [xxd(1)](https://linux.die.net/man/1/xxd)required).
 
-### Signing the message
-Signing is easier after obtained the wallet address and imported the private key. Running
-`SIGNATURE="$(litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet signmessage "$ADDRESS" "$MESSAGE")"` is enough.
-You can also verify the result with `litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet verifymessage "$ADDRESS" "$SIGNATURE" "$MESSAGE"`.
+## Signing a message with litecoin-cli
+To sign the message `29553f9e37fa16e45f1d3e616ac5366f6afd9936477f2d6fc870f49bdf540157`, we can run
+`litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet signmessage mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2 29553f9e37fa16e45f1d3e616ac5366f6afd9936477f2d6fc870f49bdf540157`,
```
Comment (gpBlockchain) [inline docs/litecoin.md:97]:
```suggestion
```litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet signmessage mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2 29553f9e37fa16e45f1d3e616ac5366f6afd9936477f2d6fc870f49bdf540157```
```

File: docs/litecoin.md (hunk 1 above)
Comment (15168316096) [reply docs/litecoin.md:26]: Consider starting a litecoin node in the background

**Review by 15168316096 — COMMENTED** [review]
LGTM

File: docs/litecoin.md (hunk 3 above)
Comment (contrun) [reply docs/litecoin.md:107]: This is not specific to litecoin. We should use this document [ckb-auth/auth.md at main · cryptape/ckb-auth · GitHub](https://github.com/cryptape/ckb-auth/blob/main/docs/auth.md). I added a link to that.

File: docs/litecoin.md [hunk 5]
```diff
@@ -1,43 +1,99 @@
...
+  "tmweb1qqd6usenz75lvym9px3q7gakhy4a3u4ne95rwg4h3w2shm7hva28wsqemjfm7c2v7gt5sl5snf9kr6tygl3t773l6spt4cmuel4d92m038gtejwnn": {
+    "purpose": "receive"
+  }
+}
+```
+
+The legacy addresses are normally with short length (see [How is a Litecoin address generated?](https://bitcoin.stackexchange.com/questions/65282/how-is-a-litecoin-address-generated) for details).
+Here both the short addresses `mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2` and `myGi4qfsqJ8kzgLEJLfdqr75PQQFyH2DSf` may
+be used to sign message, while another short address `QjpdvL4h5jnfaj1uV5ifJNUAYZTTbjgFH5` does not work.
+Which address works and which does not work may require some trial and error.
```
[the remark this answers is not in the document]
Comment (contrun) [reply docs/litecoin.md]: Judging from [address format - litecoin constants and prefixes - Bitcoin Stack Exchange](https://bitcoin.stackexchange.com/questions/62781/litecoin-constants-and-prefixes), we should use the addresses started with `m`, `n` or `L`. I didn't find any official documentation for this.

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**